src/topology/fanout.rs

```diff
@@ -192,20 +191,20 @@ impl Sink<Event> for Fanout {
 #[cfg(test)]
 mod tests {
     use super::{ControlMessage, Fanout};
-    use crate::{sink::BoundedSink, test_util::collect_ready, Event};
+    use crate::{test_util::collect_ready, Event};
+    use futures::channel::mpsc;
     use futures::{stream, Sink, SinkExt, StreamExt};
     use std::{
         pin::Pin,
         task::{Context, Poll},
     };
-    use tokio::sync::mpsc;
-    use tokio::time::{delay_for, Duration};
+    use tokio::time::{sleep, Duration};
 
     #[tokio::test]
     async fn fanout_writes_to_all() {
-        let (tx_a, rx_a) = unbounded_channel();
+        let (tx_a, rx_a) = mpsc::unbounded();
         let tx_a = Box::new(tx_a.sink_map_err(|_| unreachable!()));
-        let (tx_b, rx_b) = unbounded_channel();
+        let (tx_b, rx_b) = mpsc::unbounded();
         let tx_b = Box::new(tx_b.sink_map_err(|_| unreachable!()));
 
         let mut fanout = Fanout::new().0;
```

Comment:
Found the bug: this is dropping the control channel immediately, causing it to panic when being polled.

Reply:
Fixed in 873a081.

Reply:
This behavior should not lead to crashes anymore after 1508eb4.